src/forms/cashflow.js

```diff
@@ -42,68 +43,95 @@ const CashFlowContainer = function ({ children, label, validRow, message }) {
  *
- * @function
  * @param {object} props
  * @property {object} props.generic - Base name for the client property that
  *     needs to be updated (now the code has changed, this may be a misnomer)
  * @property {object} props.timeState - Client, either future values or current values
  * @property {object} props.updateClientValue - Updates client state
  * @property {object} props.children - Text for the row label
- *
- * @returns Component
+ * 
  */
 /** @todo Find elegant way to combine CashFlowInputsRow and MonthlyCashFlowRow
       use `includes` array to include only certain columns perhaps. */
-const CashFlowInputsRow = function ({ generic, timeState, updateClientValue, children }) {
-
-  var updateClient = function (evnt, inputProps, data) {
-    var monthly = toMonthlyAmount[ data.interval ](evnt, inputProps.value),
-        obj     = { name: generic, value: monthly };
-    updateClientValue(evnt, obj);
-  };
-
-  /** baseVal
-   * Get the time ('future' or 'current') monthly value unless there is
-   *     none, in which case, get the 'current' monthly cash flow value
-   *     (to prefill future values with 'current' ones if needed).
-   *
-   * @var
-   *
-   * @todo Add some kind of UI indication when it's the same as the 'current'
-   *     value. What if some of the row's values are the same and some are
-   *     different?
-   */
-  var baseVal   = timeState[ generic ],
-      baseProps = {
-        name:             generic,
-        className:        'cashflow-column',
-        store:            updateClient,
-        displayValidator: hasOnlyNonNegNumberChars,
-        storeValidator:   isNonNegNumber,
-        format:           toMoneyStr,
-      };
-
-  return (
-    <CashFlowContainer
-      label={ children }
-      validRow={ true }
-      message={ null }>
-      <ManagedNumberField
-        { ...baseProps }
-        value     = { baseVal / (4 + 1 / 3) }
-        otherData = {{ interval: 'weekly' }} />
-      <ManagedNumberField
-        { ...baseProps }
-        value     = { baseVal }
-        otherData = {{ interval: 'monthly' }} />
-      <ManagedNumberField
-        { ...baseProps }
-        value     = { baseVal * 12 }
-        otherData = {{ interval: 'yearly' }} />
-    </CashFlowContainer>
-  );
-
-};  // End CashFlowInputsRow{} Component
+class CashFlowInputsRow extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { message:'' }
+    // Bind the validator to set message
+    this.cashFlowStoreValidator = this.cashFlowStoreValidator.bind(this);
+  }
+
+  // Special store validator that handles maximums and sets error message
+  cashFlowStoreValidator(max) {
+    return function (str) {
+      if(!isNonNegNumber) {
+        this.setState({ message: 'Invalid number format' });
+        return false;
+      }
+      else if(parseFloat(str) > max) {
+        this.setState({ message: 'The input number exceeds the maximum of $999,999.99/yr' });
```

Comment:
It would be nice to use a template string to insert the actual maximum value in the string here, assuming we're going to store it in a variable. For example: `` `The input number exceeds the maximum of $${ maxYearly }/yr` ``